src/lightning/pytorch/utilities/testing/_runif.py

```diff
@@ -0,0 +1,199 @@
+from lightning.pytorch.core.module import _ONNX_AVAILABLE
+from lightning.pytorch.strategies.deepspeed import _DEEPSPEED_AVAILABLE
+from lightning.pytorch.utilities.imports import _OMEGACONF_AVAILABLE, _PSUTIL_AVAILABLE
+
+if module_available("_pytest"):
+    from _pytest.mark import MarkDecorator
+else:
+    MarkDecorator = Any
+
+_SKLEARN_AVAILABLE = RequirementCache("scikit-learn")
+
+
+class _RunIf:
+    """Wrapper around ``pytest.mark.skipif`` with specific conditions.
+
+    Example:
+
+        @RunIf(min_python="3.6")
+        @pytest.mark.parametrize("arg1", [1, 2.0])
+        def test_wrapper(arg1):
+            assert arg1 > 0.0
+    """
+
+    def __new__(
+        self,
+        min_cuda_gpus: int = 0,
+        min_torch: Optional[str] = None,
+        max_torch: Optional[str] = None,
+        min_python: Optional[str] = None,
+        bf16_cuda: bool = False,
+        tpu: bool = False,
+        ipu: bool = False,
+        hpu: bool = False,
+        mps: Optional[bool] = None,
+        skip_windows: bool = False,
+        standalone: bool = False,
+        deepspeed: bool = False,
+        rich: bool = False,
+        omegaconf: bool = False,
+        psutil: bool = False,
+        sklearn: bool = False,
+        onnx: bool = False,
+    ) -> MarkDecorator:  # not the real return because it would require that pytest is available
+        """Configure.
+
+        Args:
+            min_cuda_gpus: Require this number of gpus and that the ``PL_RUN_CUDA_TESTS=1`` environment variable is set.
+            min_torch: Require that PyTorch is greater or equal than this version.
+            max_torch: Require that PyTorch is less than this version.
+            min_python: Require that Python is greater or equal than this version.
+            bf16_cuda: Require that CUDA device supports bf16.
+            tpu: Require that TPU is available.
+            ipu: Require that IPU is available and that the ``PL_RUN_IPU_TESTS=1`` environment variable is set.
+            hpu: Require that HPU is available.
+            mps: If True: Require that MPS (Apple Silicon) is available,
+                if False: Explicitly Require that MPS is not available
+            skip_windows: Skip for Windows platform.
+            standalone: Mark the test as standalone, our CI will run it in a separate process.
+                This requires that the ``PL_RUN_STANDALONE_TESTS=1`` environment variable is set.
+            deepspeed: Require that microsoft/DeepSpeed is installed.
+            rich: Require that willmcgugan/rich is installed.
+            omegaconf: Require that omry/omegaconf is installed.
+            psutil: Require that psutil is installed.
+            sklearn: Require that scikit-learn is installed.
+            onnx: Require that onnx is installed.
+        """
+        import pytest
+
+        conditions = []
+        reasons = []
+        kwargs: dict = {}  # ToDo: this is very confusing, drop it
```

Comment:
We cannot drop this. It's required for our CI to run under the different `PL_RUN_X_TESTS`

Reply:
hmm so lets at least better explain why it is needed :)

Reply:
See my comment in the conversation above. It's used in https://github.com/Lightning-AI/lightning/blob/072ef26309abc0beb1a42f2b1352c8225f316fef/tests/tests_fabric/conftest.py#L143-L149

Reply:
```suggestion
        kwargs: dict = {}  # It's required for our CI to run under the different PL_RUN_X_TESTS
```